src/ucp/core/ucp_ep.c

```diff
@@ -2958,5 +2963,41 @@ ucs_status_t ucp_ep_query(ucp_ep_h ep, ucp_ep_attr_t *attr)
 #endif
     }
 
+    if (!(attr->field_mask & UCP_EP_ATTR_FIELD_LOCAL_SOCKADDR) &&
+        !(attr->field_mask & UCP_EP_ATTR_FIELD_REMOTE_SOCKADDR)) {
+        return UCS_OK;
+    }
+
+    if (uct_cm_ep == NULL) {
+        ucs_debug("uct_cm_ep is NULL");
+        return UCS_ERR_NOT_CONNECTED;
+    }
+
+    memset(&uct_cm_ep_attr, 0, sizeof(uct_ep_attr_t));
+
+    if (attr->field_mask & UCP_EP_ATTR_FIELD_LOCAL_SOCKADDR) {
+        uct_cm_ep_attr.field_mask |= UCT_EP_ATTR_FIELD_LOCAL_SOCKADDR;
+    }
+
+    if (attr->field_mask & UCP_EP_ATTR_FIELD_REMOTE_SOCKADDR) {
+        uct_cm_ep_attr.field_mask |= UCT_EP_ATTR_FIELD_REMOTE_SOCKADDR;
+    }
+
+    status = uct_ep_query(uct_cm_ep, &uct_cm_ep_attr);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    if (uct_cm_ep_attr.field_mask & UCT_EP_ATTR_FIELD_LOCAL_SOCKADDR) {
+        ucp_sockaddr_copy_always((struct sockaddr *)&attr->local_sockaddr,
+                                 (struct sockaddr *)&uct_cm_ep_attr.local_address);
+    }
+    
+    if (uct_cm_ep_attr.field_mask & UCT_EP_ATTR_FIELD_REMOTE_SOCKADDR) {
+        ucp_sockaddr_copy_always((struct sockaddr *)&attr->remote_sockaddr,
+                                 (struct sockaddr *)&uct_cm_ep_attr.remote_address);
+    }
```

Comment:
let's put the sockaddr query code in a separate function:
```
...
if (attr->field_mask & (LOCAL_SOCKADDR|REMOTE_SOCKADDR)) {
   status = ucp_ep_query_sockaddr(ep, attr);
   if (status != UCS_OK) {
     return status;
   }
}

return UCS_OK;
```

Reply:
@yosefe Done